api/server/server.go

```diff
@@ -119,6 +121,11 @@ func (s *HTTPServer) Close() error {
 
 func (s *Server) makeHTTPHandler(handler httputils.APIFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		// Daemon is abort to shutdown, server can't do api requets any more.
+		if s.listenerClosed {
```

Comment:
We can't do this because this is async. This could use atomic or a mutex ... but really is it worth it?
Maybe instead we can inherit a context which is cancelled on shutdown?

Reply:
@cpuguy83 I'm not sure if I have got what you means,PTAL , thanks.

Reply:
I think in order to get the desired result, we'll need to refactor how the server is started... namely we need to pass in a context that is tied to the daemon lifecycle so that it can be propagated to the request context.